runtime-k3d/src/gen_helm.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (c) 2023 Robert Bosch GmbH and Microsoft Corporation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import yaml
+from yaml.loader import SafeLoader
+from typing import Optional
+import os
+
+from lib import (
+    get_services,
+    get_script_path,
+    create_nodeport_spec,
+    create_cluster_ip_spec,
+    get_template
+)
+
+node_port = 30050
+
+def find_service_spec_index(lst, name):
+    """Returns index of spec for the given name.
+    Args:
+        lst: The list of the template specifications.
+        name: The name of the pod.
+    """
+    for i, elem in enumerate(lst):
+        dict_keys = list(lst[i].keys())
+        if elem[dict_keys[0]]['name'] == name:
+            return i    
+
+
+def generate_values_and_templates(values_template, service_spec):
+    """Generates values specification from the given templates and service specification and generates helm templates for each service spec.
+    Args:
+        lst: The list of the template specifications.
+        kind: The kind of the pod.
+        name: The name of the pod.
+    """
+    global node_port    
+    service_id = service_spec["id"]
+    
+    no_dapr=False
+    container_image = None
+    service_port = None
+    env_vars = dict[str, Optional[str]]()
+    port_forwards = []
+    mounts = []
+    args = []
+
+    for config_entry in service_spec["config"]:
+        if config_entry["key"] == "image":
+            container_image = config_entry["value"]
+        elif config_entry["key"] == "env":
+            pair = config_entry["value"].split("=", 1)
+            env_vars[pair[0].strip()] = None
+            if len(pair) > 1:
+                env_vars[pair[0].strip()] = pair[1].strip()
+        elif config_entry["key"] == "port":
+            service_port = config_entry["value"]
+        elif config_entry["key"] == "no-dapr":
+            no_dapr = config_entry["value"]
+        elif config_entry["key"] == "arg":
+            args.append(config_entry["value"])
+        elif config_entry["key"] == "port-forward":
+            port_forwards.append(config_entry["value"])
+        elif config_entry["key"] == "mount":
+            mounts.append(config_entry["value"])
```

Comment:
please use switch case

Reply:
Python doesn't have switch statement.

Reply:
since python 3.10 it has: https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching

ok it is called "match case" :)

Reply:
But, should we change it if it is only support in Python 3.10 and greater?

Reply:
since we are using 3.10 in the template and the sdk I don't see any problem.